### DISCUSSION TITLE
SVG TO ICO, 24bit

### DISCUSSION THREAD

**Discussion by vdolbakyan** [ImageMagick/ImageMagick#2099] [Help]
Hello,

I am new to imagemagick
I  am trying to create ico file with multiple sizes, using following command line, but when I open ico file in Visual Studio each icon shows up as 32 bit and as a result it can not be edited in Visual Studio, because VS icon editot only supports 24 bit. Is there way to force 24 bit icons ?

Also is there a good guide how to use -density flag when converting from svg?

```

ImageMagick-7.0.10-Q16\magick.exe" convert -background transparent -density 384  -define icon:auto-resize=20,24,32,48,64,72,96,128,256 -verbose "x.svg" -depth 24 "x.ico"
x.svg SVG 488x380 488x380+0+0 16-bit sRGB 925B 0.016u 0:00.015
x.svg=> PNG 488x380=>256x256 256x256+0+0 24-bit sRGB 0.125u 0:00.032
x.svg=>x.ico SVG 488x380 488x380+0+0 24-bit sRGB 194074B 0.422u 0:00.385

```

Thank you
Vigen

**Comment by fmw42** [comment]
In IM 7 use magick, not magick convert. You cannot have transparency if you want 24-bit results.  So you would have to remove the transparency or flatten the image against some color. Try

```
magick.exe -verbose -density 384 "x.svg" ^
-define icon:auto-resize=20,24,32,48,64,72,96,128,256 ^
-alpha off -type truecolor "x.ico"
```

**Reply by vdolbakyan** [reply]
Thank you for a quick respose 
However i lost  transparency and Visual Studio still thinks this 32 bit ico. (output is attached)

```
"magick.exe" -verbose -background transparent -density 384  "x.svg" -define icon:auto-resize=20,24,32,48,64,72,96,128,256 -type truecolor  "x.ico"
x.svg SVG 488x380 488x380+0+0 16-bit sRGB 925B 0.016u 0:00.013
x.svg=> PNG 488x380=>256x256 256x256+0+0 16-bit TrueColor sRGB 0.063u 0:00.016
x.svg=>x.ico SVG 488x380 488x380+0+0 16-bit TrueColor sRGB 200302B 0.469u 0:00.401
```

[x.zip](https://github.com/ImageMagick/ImageMagick/files/4725006/x.zip)

**Reply by fmw42** [reply]
You did not use -alpha off, though I doubt that will make a difference. But try it anyway. 

Sorry, but I am not on Windows and do not use ICO files. So I do not know why VS still thinks it is 32-bits. ImageMagick shows it as 24-bits when it says TruColor.  Perhaps it has to do with being 16-bits.  Try adding -depth 8 just before -type truecolor.  See if that helps.  Also remove -background transparent.
